larpandoracontent/LArTrackShowerId/TrackShowerIdFeatureTool.cc

```diff
@@ -326,19 +373,781 @@ StatusCode MipEnergyFeatureTool::ReadSettings(const TiXmlHandle xmlHandle)
 //------------------------------------------------------------------------------------------------------------------------------------------
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+VertexDistanceFeatureTool::VertexDistanceFeatureTool() :
+    m_ratioVariables(true),
+    m_slidingLinearFitWindow(10000)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
 void VertexDistanceFeatureTool::Run(SupportVectorMachine::DoubleVector &featureVector, const Algorithm *const pAlgorithm,
     const pandora::Cluster *const pCluster)
 {
     if (PandoraContentApi::GetSettings(*pAlgorithm)->ShouldDisplayAlgorithmInfo())
         std::cout << "----> Running Algorithm Tool: " << this->GetInstanceName() << ", " << this->GetType() << std::endl;
 
-    featureVector.push_back(CutClusterCharacterisationAlgorithm::GetVertexDistance(pAlgorithm, pCluster));
+    if (m_ratioVariables)
+    {
+        float straightLineLength(-1.f), ratio(-1.f);
+        try
+        {
+            const TwoDSlidingFitResult slidingFitResultLarge(pCluster, m_slidingLinearFitWindow, LArGeometryHelper::GetWireZPitch(this->GetPandora()));
+            straightLineLength = (slidingFitResultLarge.GetGlobalMaxLayerPosition() - slidingFitResultLarge.GetGlobalMinLayerPosition()).GetMagnitude();
+            if (straightLineLength > std::numeric_limits<double>::epsilon())
+                ratio = (CutClusterCharacterisationAlgorithm::GetVertexDistance(pAlgorithm, pCluster))/straightLineLength;
+        }
+        catch (const StatusCodeException &)
+        {
+            ratio = -1.f;
+        }
+        featureVector.push_back(ratio);
+    }
+    else
+    {
+        featureVector.push_back((CutClusterCharacterisationAlgorithm::GetVertexDistance(pAlgorithm, pCluster)));
+    }
+}
```

Comment:
See above comments. An important question (similar to discussion above) is the expectation for use with and without ratio variables. Is it worth supporting ratio variables on/off if this will only ever be used in one configuration? The ratioVarables option certainly has a big footprint in the implementation!

Reply:
under discussion as well